src/main/java/tc/oc/pgm/map/NameCacheUtil.java

```diff
@@ -0,0 +1,122 @@
+package tc.oc.pgm.map;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.stream.JsonWriter;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.UUID;
+import org.bukkit.Bukkit;
+import tc.oc.pgm.api.PGM;
+
+public class NameCacheUtil {
+
+  private static final Gson GSON = new Gson();
+
+  private static List<UUID> unresolved = new ArrayList<UUID>();
+  private static HashMap<UUID, CachedPlayer> cache = new HashMap<UUID, CachedPlayer>();
+
+  public static boolean isUUIDCached(UUID uuid) {
+    return cache.containsKey(uuid);
+  }
+
+  public static CachedPlayer getCachedPlayer(UUID uuid) {
+    return cache.get(uuid);
+  }
+
+  public static void addUUID(UUID uuid) {
+    if (!cache.containsKey(uuid)) {
+      unresolved.add(uuid);
+    }
+  }
+
+  private static String resolveName(UUID uuid) throws IOException {
+    HttpURLConnection con =
+        (HttpURLConnection)
+            new URL("https://api.ashcon.app/mojang/v2/user/" + uuid.toString()).openConnection();
+    con.setRequestMethod("GET");
+    con.setRequestProperty("Accept", "application/json");
+    con.setConnectTimeout(5000);
+    con.setReadTimeout(5000);
+    BufferedReader br =
+        new BufferedReader(new InputStreamReader(con.getInputStream(), StandardCharsets.UTF_8));
+    StringBuilder response = new StringBuilder();
+    String line;
+    while ((line = br.readLine()) != null) {
+      response.append(line.trim());
+    }
+    br.close();
+    return GSON.fromJson(response.toString(), JsonObject.class).get("username").getAsString();
+  }
+
+  public static void resolveAll() {
+    Bukkit.getScheduler()
+        .runTaskAsynchronously(
+            PGM.get(),
+            () -> {
+              for (int i = 0; i < unresolved.size(); i++) {
```

Comment:
```suggestion
              for (UUID uuid : unresolved) {
```

There's no reason to iterate with a for-i loop, and you won't be able to do that after  you use a set